cognite/client/data_classes/agents/chat.py

```diff
@@ -85,6 +86,250 @@ def _load_content(cls, data: dict[str, Any]) -> UnknownContent:
+    """
+
+    _type: ClassVar[str] = "toolConfirmation"
+    action_id: str
+    content: MessageContent
+    tool_name: str
+    tool_arguments: dict[str, object]
+    tool_description: str
+    tool_type: str
+    details: dict[str, object] | None = None
+
+    def dump(self, camel_case: bool = True) -> dict[str, Any]:
+        result: dict[str, Any] = {
+            "type": self._type,
+            "actionId" if camel_case else "action_id": self.action_id,
+            self._type: {
+                "content": self.content.dump(camel_case=camel_case),
+                "toolName" if camel_case else "tool_name": self.tool_name,
+                "toolArguments" if camel_case else "tool_arguments": self.tool_arguments,
+                "toolDescription" if camel_case else "tool_description": self.tool_description,
+                "toolType" if camel_case else "tool_type": self.tool_type,
+            },
+        }
+        if self.details is not None:
+            result[self._type]["details"] = self.details
+        return result
+
+    @classmethod
+    def _load_call(cls, data: dict[str, Any], cognite_client: CogniteClient | None = None) -> ToolConfirmationCall:
+        tool_confirmation = data[cls._type]
+        content = MessageContent._load(tool_confirmation["content"])
+        return cls(
+            action_id=data["actionId"],
+            content=content,
+            tool_name=tool_confirmation["toolName"],
+            tool_arguments=tool_confirmation["toolArguments"],
+            tool_description=tool_confirmation["toolDescription"],
+            tool_type=tool_confirmation["toolType"],
+            details=tool_confirmation.get("details"),
+        )
+
+
+@dataclass
+class UnknownActionCall(ActionCall):
+    """Unknown action call type for forward compatibility.
+
+    Args:
+        action_id (str): The unique identifier for this action call.
+        type (str): The action call type.
+        data (dict[str, object]): The raw action call data.
+    """
+
+    action_id: str
+    type: str
+    data: dict[str, object] = field(default_factory=dict)
+
+    def dump(self, camel_case: bool = True) -> dict[str, Any]:
+        result = self.data.copy()
+        result["type"] = self.type
+        result["actionId" if camel_case else "action_id"] = self.action_id
+        return result
+
+    @classmethod
+    def _load_call(cls, data: dict[str, Any], cognite_client: CogniteClient | None = None) -> UnknownActionCall:
+        action_type = data["type"]
+        action_id = data["actionId"]
+        return cls(action_id=action_id, data=data, type=action_type)
+
+
+# Build the mapping AFTER concrete classes are defined
+_ACTION_CALL_CLS_BY_TYPE: dict[str, type[ActionCall]] = {
+    subclass._type: subclass  # type: ignore[type-abstract]
+    for subclass in ActionCall.__subclasses__()
+    if hasattr(subclass, "_type") and not getattr(subclass, "__abstractmethods__", None)
```

Comment:
Just let this fail if missing attr "_type", then the person implementing the new subclass will immediately know to fix it (if mypy isn't already unhappy).

Also, it's better to use something like `inspect.isabstract` to check for and exclude abstract subclasses. In this case, I would remove it. If you ever are to add such a subclass in the future, then you can add it 😅 
```suggestion
    if subclass is not Unknown(...)
```